### PR TITLE
ref(js): Upgrade mobx v5 -> v6

### PR DESCRIPTION
**Pull request by evanpurkhiser**
This is a relatively trivial update since we don't use mobx much. The largest changes here are

 - You now call makeObservable on any objects that have observables / computes / actions.

   They changed this due to changes in how javascript decorators work.  We could actually completely drop decorators here if we wanted to, but I'll leave them for now and use the `makeObservable(this)` incantation.

 - toJSON on ObservableMap has changed to return an entries list. We remedy this using Object.fromEntries to get back the object representation like before

 - The `loose: true` option on property assignment is no longer needed.  I added this in c7da02e7f055ecf5bd135ab542e729096427bc7f where I upgrade to babel v7. We already had mobx at this point so I am relatively sure I did this to keep mobx compatibility.

   Ran into some wonkyness with our storybook build because of this. See https://github.com/babel/babel/issues/11622#issuecomment-638609015

### CODE REVIEW

File: src/sentry/static/sentry/app/views/monitors/monitorModel.tsx [hunk 1]
```diff
@@ -8,34 +8,31 @@ type TransformedData = {
 
 export default class MonitorModel extends FormModel {
   getTransformedData() {
-    return Object.entries(this.fields.toJSON()).reduce<TransformedData>(
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/monitors/monitorModel.tsx:11]: This is just a whitespace change.

The real diff is

```diff
- Object.entries(this.fields.toJSON())
+ this.fields.toJSON()
```

File: src/sentry/static/sentry/app/views/settings/components/forms/formField/index.tsx [hunk 2]
```diff
@@ -367,12 +367,16 @@ class FormField<P extends {} = {}> extends React.Component<Props<P>> {
...
+                const isVisible =
+                  typeof props.visible === 'function'
                     ? props.visible({...this.props, ...props} as ResolvedProps<P>)
-                    : true) &&
-                    selectionInfoFunction({...props, error, value})) ||
-                  null
+                    : true;
+
+                return (
+                  <React.Fragment>
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/components/forms/formField/index.tsx:377]: Types changed here slightly to require a `React.ReactElement` return type

**Comment by EvanPurkhiser** [comment]
Hm, I have the storybook thing fixed locally

**Comment by EvanPurkhiser** [comment]
Feel free to verify in the preview build

**Comment by billyvg** [comment]
Hmm so even if we're using `<Observer>`, you have to use `makeObservable()`?

**Comment by EvanPurkhiser** [comment]
@billyvg the `makeObservable` addition is needed because of the changes to the standardized version of decorators

> MobX before version 6 encouraged the use of ES.next decorators to mark things as observable, computed and action. However, decorators are currently not an ES standard, and the process of standardization is taking a long time. It also looks like the standard will be different from the way decorators were implemented previously. In the interest of compatibility we have chosen to move away from them in MobX 6, and recommend the use of makeObservable / makeAutoObservable instead.

So you can call `makeObservable(this)` and it will read decorator metadata

> MobX before version 6 did not require the makeObservable(this) call in the constructor, but because it makes the implementation of decorator simpler and more compatible, it now does. This instructs MobX to make the instances observable following the information in the decorators -- the decorators take the place of the second argument to makeObservable.

Just copying all this from here https://mobx.js.org/enabling-decorators.html

The `<Observable />` component is just used to react to our observables, which hasn't changed.

**Review by billyvg — APPROVED** [review]

**Pull request merged**